services/endpointsrv/service.go

```diff
@@ -332,164 +311,6 @@ func verifyCanRunChecks(endpoint types.Endpoint, evals []types.Eval) error {
 	return nil
 }
 
-func buildEndpointCheckSteps(ctx context.Context, checkID string, store Store, endpoint types.Endpoint, evals []types.Eval) ([]checkEndpointStep, error) {
-	var checks []checkEndpointStep
-
-	for _, eval := range evals {
-		datasetPath := "https://" + eval.HTTPEndpoint + "/dataset"
-		assertPath := "https://" + eval.HTTPEndpoint + "/assert"
-
-		d, err := fetchDataset(datasetPath)
-		if err != nil {
-			return nil, fmt.Errorf("fetch dataset: %w", err)
-		}
-
-		for _, item := range d.Data {
-			stepID := typeid.Must(typeid.New("step")).String()
-
-			checks = append(checks, checkEndpointStep{
-				checkID:    checkID,
-				stepID:     stepID,
-				input:      item.Input,
-				assertPath: assertPath,
-				endpoint:   "https://" + endpoint.HTTPAddress,
-				store:      store,
-			})
-
-			if err := store.EndpointCheckStepCreate(ctx, db.EndpointCheckStepCreateParams{
-				ID:      stepID,
-				CheckID: checkID,
-				EvalID:  eval.ID,
-				Input:   sql.NullString{String: string(item.Input), Valid: true},
-			}); err != nil {
-				return nil, fmt.Errorf("create check step: %w", err)
-			}
-		}
-	}
-
-	return checks, nil
-}
-
-type checkEndpointStep struct {
-	err error
-
-	checkID          string
-	stepID           string
-	assertPath       string
-	endpoint         string
-	input            json.RawMessage
-	endpointResponse json.RawMessage
-	assertion        string
-	store            Store
-}
-
-func (c *checkEndpointStep) callEndpoint(ctx context.Context) {
```

Comment:
Can we keep the old flow as well. After what we discussed this morning, the easiest way forward is to have both flow and test which is best as we go along.

Reply:
We still have the old flow, I just moved it to `checks.go` to have all the checker logic together in a single file rather than mixed up with `service.go`

Reply:
Om great! Looks good to merge then